### PR TITLE
Octopetshop docker

### PR DESCRIPTION
**Pull request by twerthi**
Adding components to run this sample application in containers.

### CODE REVIEW

File: OctopusSamples.OctoPetShop.Database/dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM mcr.microsoft.com/dotnet/core/sdk:2.1
+
+RUN mkdir /src
+WORKDIR /src
+ADD . /src
+RUN dotnet restore
+RUN ["dotnet", "build", "--configuration", "release"]
+
+ENTRYPOINT [ "dotnet", "run", "--no-launch-profile", "Data Source=192.168.1.4;Initial Catalog=OctoPetShop; User ID=sa; Password=Great balls 0f fire!" ]
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.Database/dockerfile]: Can you make the connection string configurable?

File: OctopusSamples.OctoPetShop.ProductService/Startup.cs [hunk 2]
```diff
@@ -36,7 +36,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 app.UseHsts();
             }
 
-            app.UseHttpsRedirection();
+            //app.UseHttpsRedirection();
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.ProductService/Startup.cs]: Can you make this configurable too?

File: OctopusSamples.OctoPetShop.ProductService/appsettings.json [hunk 3]
```diff
@@ -10,6 +10,6 @@
     "EnvironmentName": "DEV"
   },
   "ConnectionStrings": {
-    "OPSConnectionString": "Server=(local)\\SqlExpress; Database=ops; Trusted_connection=true"
+    "OPSConnectionString": "Data Source=192.168.1.4;Initial Catalog=OctoPetShop; User ID=sa; Password=Great balls 0f fire!"
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.ProductService/appsettings.json]: I think we should leave this with the previous default value.

File: OctopusSamples.OctoPetShop.ShoppingCartService/appsettings.json [hunk 4]
```diff
@@ -10,6 +10,6 @@
     "EnvironmentName": "DEV"
   },
   "ConnectionStrings": {
-    "OPSConnectionString": "Server=(local)\\SqlExpress; Database=ops; Trusted_connection=true"
+    "OPSConnectionString": "Data Source=192.168.1.4;Initial Catalog=OctoPetShop; User ID=sa; Password=Great balls 0f fire!"
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.ShoppingCartService/appsettings.json]: I think we should leave this with the previous default value

File: OctopusSamples.OctoPetShop.ProductService/dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+WORKDIR /src
+ADD . /src
+RUN dotnet restore
+RUN ["dotnet", "build", "--configuration", "release"]
+
+
+ENV ASPNETCORE_URLS="http://+:5011;https://+:5014"
+ENV ASPNETCORE_ENVIRONMENT="Production"
+
+ENTRYPOINT [ "dotnet", "run", "--no-launch-profile" ]
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.ProductService/dockerfile]: Does this need to expose a port?

File: OctopusSamples.OctoPetShop.Web/appsettings.json [hunk 6]
```diff
@@ -10,7 +10,7 @@
   "AppSettings": {
     "AppVersion": "0.0.0",
     "EnvironmentName": "DEV",
-    "ProductServiceBaseUrl": "http://localhost:52132/",
-    "ShoppingCartServiceBaseUrl": "https://localhost:52131/"
+    "ProductServiceBaseUrl": "http://192.168.1.1:5011/",
+    "ShoppingCartServiceBaseUrl": "http://192.168.1.3:5012"
   }
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.Web/appsettings.json]: Leave the default values

File: OctopusSamples.OctoPetShop.Web/dockerfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+WORKDIR /src
+ADD . /src
+RUN dotnet restore
+RUN ["dotnet", "build", "--configuration", "release"]
+
+
+ENV ASPNETCORE_URLS="http://+:5000;https://+:5001"
+ENV ASPNETCORE_ENVIRONMENT="Production"
+
+ENTRYPOINT [ "dotnet", "run", "--no-launch-profile" ]
```
Comment (ryanrousseau) [inline OctopusSamples.OctoPetShop.Web/dockerfile]: Does this need to expose ports?

File: OctopusSamples.OctoPetShop.Database/dockerfile (hunk 1 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.Database/dockerfile]: Added environment variable overrides so they could be specified in either the dockerfile or the docker-compose

File: OctopusSamples.OctoPetShop.ProductService/Startup.cs (hunk 2 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.ProductService/Startup.cs]: Added detection for when running in a container

File: OctopusSamples.OctoPetShop.ProductService/appsettings.json (hunk 3 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.ProductService/appsettings.json]: Environment variables now allow for connection string override.  The docker-compose file has these environment variables included so I've put the connection string back to the way it was.  Project will now run from VS as well as docker without having to make changes.

File: OctopusSamples.OctoPetShop.ShoppingCartService/appsettings.json (hunk 4 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.ShoppingCartService/appsettings.json]: Done.

File: OctopusSamples.OctoPetShop.ProductService/dockerfile (hunk 5 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.ProductService/dockerfile]: I think the docker-compose file may have been doing this for me, which was why I didn't think it's necessary.  Added the EXPOSE directive to all docker files.

File: OctopusSamples.OctoPetShop.Web/appsettings.json (hunk 6 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.Web/appsettings.json]: Put back to default

File: OctopusSamples.OctoPetShop.Web/dockerfile (hunk 7 above)
Comment (twerthi) [reply OctopusSamples.OctoPetShop.Web/dockerfile]: EXPOSE directive added

**Pull request merged**